walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,68 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.Timer;
+import java.util.stream.Collectors;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    long duration = request.getDuration();
+    // The duration should not be longer than 1 day
+    if (duration > TimeRange.getTimeInMinutes(23, 59)) {
+      return Arrays.asList();
+    }
+    Collection<String> attendees = request.getAttendees();
+    List<TimeRange> busyTimes = new ArrayList<>();
+
+    // Find the other events the attendees are already scheduled in for.
+    for (Event event : events) {
+      Set<String> intersection = attendees.stream()
+                                     .distinct()
+                                     .filter(event.getAttendees()::contains)
+                                     .collect(Collectors.toSet());
+      if (intersection.size() > 0) {
+        busyTimes.add(event.getWhen());
+      }
+    }
+
+    busyTimes.sort(TimeRange.ORDER_BY_START);
+    List<TimeRange> mergedBusyTimes = new ArrayList<>();
+    for (TimeRange timeRange : busyTimes) {
```

Comment:
how does naming the variable _timeRange_ helping us understand the code.

Reply:
Changed the name to unavailableTimeRange to help clarity

Reply:
Thanks